src/python/pants/engine/internals/build_files.py

```diff
@@ -93,6 +94,7 @@ async def evaluate_preludes(
     )
     globals: dict[str, Any] = {
         **{name: getattr(builtins, name) for name in dir(builtins) if name.endswith("Error")},
+        **{name: getattr(typing, name) for name in dir(typing)},
```

Comment:
`dir` on a module appears to include all of the top-level names including any imported modules. For `typing` this includes `sys` (and `__builtins__`). Is that intentional?